UofT-Frosh-Orientation/orientation-website
# 608 load up landing page

## **Contribution:**

- Looping through the created landing pages (currently Tanu, Uzma and mine from last year), to render a different landing page on every refresh using local storage keys.
- added wrapper with variable `readyForFrosh` to control what is displayed

### **Issue:**

Closes #608 

### **Accomplishments:**

- set up local storage key `landing_page_idx` (will show up in dev tools F12 tab under Application > Local Storage > Key)
- generates a "random" local storage key each time, which indexes a certain landing page in the defined array

### **Visuals:** 

- Shows all landing pages from previous issues #598 

## **Details:**

### **Expected Behavior:**

- upon first loading, you'll see one landing page, and after refreshing you'll see a different landing page each time
- landing pages should look exactly the same as the ones from the prs

### **Testing Steps:**

Steps required to get expected behavior.

- run the frontend (no need to use docker compose)
- landing page will show up as it's rendered in App
- refresh to check for different landing pages

### **Complications:**

- needed to learn how to use local storage again
- needed to alter useEffect to prevent rerenders when the local storage key was updated
- called components incorrectly and got an error `Functions are not valid as a React child`
- arrays need keys or eslint throws error
- figuring out how to generate random numbers 
- react is a little funky with indexing arrays -- had to use `map()`, if anyone else has ideas let me know

### **Resources:**

Any additional resources used outside this in issue.

- [resource 1](https://javascript.info/localstorage): docs on local storage
- [resource 2](https://github.com/UofT-Frosh-Orientation/orientation-website/blob/dev/client/src/pages/ScuntJudges/ScuntJudges.jsx): code example from code base on local storage
- [resource 3](https://stackoverflow.com/questions/4959975/generate-random-number-between-two-numbers-in-javascript): generate a random number between two given numbers
- [resource 4](https://bobbyhadz.com/blog/react-functions-are-not-valid-as-react-child): take a look for this error `Functions are not valid as a React child`
